app/editor/components/FloatingToolbar.tsx

```diff
@@ -146,54 +152,54 @@ function usePosition({
   }
 }
 
-function FloatingToolbar(props: Props) {
-  const menuRef = props.forwardedRef || React.createRef<HTMLDivElement>();
-  const [isSelectingText, setSelectingText] = React.useState(false);
+const FloatingToolbar = React.forwardRef(
+  (props: Props, forwardedRef: React.RefObject<HTMLDivElement>) => {
+    const menuRef = forwardedRef || React.createRef<HTMLDivElement>();
+    const [isSelectingText, setSelectingText] = React.useState(false);
 
-  const position = usePosition({
-    menuRef,
-    isSelectingText,
-    props,
-  });
+    const position = usePosition({
+      menuRef,
+      isSelectingText,
+      props,
+    });
 
-  React.useEffect(() => {
-    const handleMouseDown = () => {
-      if (!props.active) {
-        setSelectingText(true);
-      }
-    };
+    React.useEffect(() => {
+      const handleMouseDown = () => {
+        if (!props.active) {
+          setSelectingText(true);
+        }
+      };
 
-    const handleMouseUp = () => {
-      setSelectingText(false);
-    };
+      const handleMouseUp = () => {
+        setSelectingText(false);
+      };
 
-    window.addEventListener("mousedown", handleMouseDown);
-    window.addEventListener("mouseup", handleMouseUp);
+      window.addEventListener("mousedown", handleMouseDown);
+      window.addEventListener("mouseup", handleMouseUp);
 
-    return () => {
-      window.removeEventListener("mousedown", handleMouseDown);
-      window.removeEventListener("mouseup", handleMouseUp);
-    };
-  }, [props.active]);
+      return () => {
+        window.removeEventListener("mousedown", handleMouseDown);
+        window.removeEventListener("mouseup", handleMouseUp);
+      };
+    }, [props.active]);
 
-  // only render children when state is updated to visible
-  // to prevent gaining input focus before calculatePosition runs
-  return (
-    <Portal>
-      <Wrapper
-        active={props.active && position.visible}
-        ref={menuRef}
-        offset={position.offset}
-        style={{
-          top: `${position.top}px`,
-          left: `${position.left}px`,
-        }}
-      >
-        {position.visible && props.children}
-      </Wrapper>
-    </Portal>
-  );
-}
+    return (
+      <Portal>
+        <Wrapper
+          active={props.active && position.visible}
+          ref={menuRef}
+          offset={position.offset}
+          style={{
+            top: `${position.top}px`,
+            left: `${position.left}px`,
+          }}
+        >
+          {props.children}
```

Comment:
Key is to always render the children otherwise the first render has the wrong width and it results in the toolbar being mispositioned off the right of the screen